server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -195,63 +231,67 @@ final class CSRF[F[_], G[_]] private[middleware] (
     }
   }
 
-  /** Embed a token into a response **/
-  def embedNew(res: Response[G]): F[Response[G]] =
-    generateToken.map(content => res.addCookie(ResponseCookie(cookieName, content)))
+  def embedInResponseCookie(r: Response[G], token: CSRFToken): Response[G] =
+    r.addCookie(createResponseCookie(token))
 
-  /** Middleware to embed a csrf token into routes that do not have one. **/
-  def withNewToken: Middleware[F, Request[G], Response[G], Request[G], Response[G]] =
-    _.andThen(embedNew _)
+  def embedInRequestCookie(r: Request[G], token: CSRFToken): Request[G] =
+    r.addCookie(createRequestCookie(token))
 
+  /** Embed a token into a response **/
+  def embedNewInCookie[M[_]: Sync](res: Response[G]): M[Response[G]] =
```

Comment:
`embedNewInResponseCookie`, for consistency?

Reply:
Will fix